ratatui-widgets/src/canvas.rs

```diff
@@ -108,67 +110,82 @@ trait Grid: fmt::Debug {
     fn reset(&mut self);
 }
 
-/// The `BrailleGrid` is a grid made up of cells each containing a Braille pattern.
+/// The `PatternGrid` is a grid made up of cells each containing a `W`x`H` pattern character.
 ///
-/// This makes it possible to draw shapes with a resolution of 2x4 dots per cell. This is useful
-/// when you want to draw shapes with a high resolution. Font support for Braille patterns is
-/// required to see the dots. If your terminal or font does not support this unicode block, you
-/// will see unicode replacement characters (�) instead of braille dots.
+/// This makes it possible to draw shapes with a resolution of e.g. 2x4 (Braille or unicode octant)
+/// per cell.
+/// Font support for the relevant pattern character is required. If your terminal or font does not
+/// support the relevant unicode block, you will see unicode replacement characters (�) instead.
 ///
-/// This grid type only supports a single foreground color for each 2x4 dots cell. There is no way
-/// to set the individual color of each dot in the braille pattern.
+/// This grid type only supports a single foreground color for each `W`x`H` pattern character.
+/// There is no way to set the individual color of each octant.
 #[derive(Debug)]
-struct BrailleGrid {
+struct PatternGrid<const W: usize, const H: usize> {
     /// Width of the grid in number of terminal columns
     width: u16,
     /// Height of the grid in number of terminal rows
     height: u16,
-    /// Represents the unicode braille patterns. Will take a value between `0x2800` and `0x28FF`
-    /// this is converted to an utf16 string when converting to a layer. See
-    /// <https://en.wikipedia.org/wiki/Braille_Patterns> for more info.
-    utf16_code_points: Vec<u16>,
+    /// Represents the pattern for each grid character in row-major order. For instance, for a 2x4
+    /// pattern marker, bits 0 to 7 should represent the following pseudo-pixels:
```

Comment:
bits seems like the wrong term here.

Reply:
I am not sure what you mean but am open to any suggestion. What I meant is that bits 0 in the `u8` represent the top left pixel etc.

Reply:
Ah got it - I assumed wrongly this was storing chars (thinking that the u8 was just a substitute for an 8 bit character). It's actually being used to store the pixels packed into the cell.

Probably add a small note about sextants / quadrants using just the first 6 / 4 bits would be nice here.

Reply:
I tried to be a bit clearer (now in `PatternCell`), let me know if this is still unclear.